### PR TITLE
[NEW] Comparing Static Site Generators

### PR DESCRIPTION
**Pull request by Guaris**

### CODE REVIEW

File: docs/websites/choose-static-site-generator.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+title: 'Choosing a Static Site Generator'
+contributor:
+  name: Ahmed Bouchefra
+  link: https://www.techiediaries.com
+---
+
+
+In this article we'll take an in-depth look at the three most popular static site generators: [Jekyll](https://jekyllrb.com/) (a Ruby generator built by Github for powering their Github pages), [Hugo](https://gohugo.io/) (an extremly fast static generator built on top of the Go programming language) and [Hexo](https://hexo.io/) (a fast website generator based on Node.js). We'll also look at more specialized options such as [Gatsby](https://www.gatsbyjs.org/) (a blazing fast static site/PWA generator built around modern web technologies such as React and GraphQL) and [Gitbook](https://www.gitbook.com/) (a CLI and Node.js library for building and hosting books using Git and Markdown).
+
+## What's Static Website Generation?
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: This is not an actual term. Use "Static site generator" instead

File: docs/websites/choose-static-site-generator.md [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+Nowadays there is a plethora of server side languages, database management systems and content management systems so why do many website owners go static?
+
+There are many reasons such as:
+
+* contents are stored as flat files so no need for a database
+* a static website doesn't need dynamic server side processing
+* static websites are super fast than dynamic websites since they require no server side processing or database access
+* static websites are more secure than any dynamic website since there are less security holes to exploit
+* super scaling when used with CDN
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: "SUPER"

File: docs/websites/choose-static-site-generator.md [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+## Reasons for Choosing the Static Approach?
+
+Nowadays there is a plethora of server side languages, database management systems and content management systems so why do many website owners go static?
+
+There are many reasons such as:
+
+* contents are stored as flat files so no need for a database
+* a static website doesn't need dynamic server side processing
+* static websites are super fast than dynamic websites since they require no server side processing or database access
+* static websites are more secure than any dynamic website since there are less security holes to exploit
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: Probably cannot say it this way

File: docs/websites/choose-static-site-generator.md [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+* super scaling when used with CDN
+* caching static files is more effecive than caching dynamic pages
+
+## Jekyll vs Hugo vs Hexo
+
+In this section we'll introduce three popular static site generators: Jekyll, Hugo and Hexo, see their pros and cons and we'll also compare them according to different criteria such as:
+
+* speed,
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: Use a table

File: docs/websites/choose-static-site-generator.md [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+
+Jekyll is by far the most popular static generator since it was built and supported by GitHub and used for the popular service GitHub Pages which are used for free to host static pages for personal or project websites.
+
+Jekyll has the largest community among other static generators that has provided a plethora of great tutorials, open source themes and plugins.
+
+It's seen as the contender of WordPress in the static world and many bloggers have migrated their blogs from WordPress to Jekyll.
+
+In StackOverflow, Jekyll has more related questions than both Hugo and Hexo
+
+![Image](https://screenshotscdn.firefoxusercontent.com/images/45ba2c16-e792-4c5f-b67a-ea4c66a324dd.png)
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: Don't do this

File: docs/websites/choose-static-site-generator.md [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+
+Hugo is not as popular as Jekyll but it's now [among the most three popular static generators](https://www.staticgen.com/) in the web. It's seen as a blazing fast alternative to Jekyll.
+
+### Speed
+
+If you’re concerned about performance and speed, Hugo is the static generator you need. It’s has many built-in features, but most importantly it's blazing fast thanks to Go.
+
+Hugo is super fast. You can build your website in terms of milliseconds (instead of seconds or even minutes in case of Jekyll)
+
+This benchmark on Youtube shows Hugo building *5000* pages in about *6* seconds
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: What benchmark?

File: docs/websites/choose-static-site-generator.md [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+
+You can easily query content from different sections and dispaly them in templates
+
+Hugo also makes it very easy to work with tags and categories. You can query all post belonging to a tag or a category with simple logic.
+
+Hugo also supports data files and dynamically loading data from remote URLs.
+
+### Plugins and Extensibility
+
+Plugins are one of the major weaknesses of Hugo. That's inherited from the fact that Go is a compiled (vs interprited) language so you have no easy and direct way to use a plugin with Hugo.
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: This needs clarification or needs to be ommitted

File: docs/websites/choose-static-site-generator.md [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+
+* Hexo has a relatively large community but the majority is non-english speakers (from China)
+
+### Community and Popularity
+
+Hexo community is getting bigger with Hexo becoming more popular. It's now among the most popular static generators in the web. The main drawback regarding Hexo community is that most of it comes from China.
+
+There are less tutorials for Hexo compared to Jekyll but the [documentation](https://hexo.io/docs/) is clear and easy to follow.
+
+As the time of this writing Hexo has *20.336* stars in Github.
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: Just no

File: docs/websites/choose-static-site-generator.md [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+* available free and open source [themes](https://hexo.io/themes/)
+* available free [plugins](https://hexo.io/plugins/)
+
+For the cons:
+
+* Hexo has a relatively large community but the majority is non-english speakers (from China)
+
+### Community and Popularity
+
+Hexo community is getting bigger with Hexo becoming more popular. It's now among the most popular static generators in the web. The main drawback regarding Hexo community is that most of it comes from China.
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: Citation needed

File: docs/websites/choose-static-site-generator.md [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+
+Gatsby is also a Progressive Web App generator so your static website is a PWA out of the box. You'll have many best practicies of modern web development that make your app the fatest possible such as code splitting, critical CSS and JavaScript and resources prefetching etc.
+
+![Image](https://screenshotscdn.firefoxusercontent.com/images/8f1bc0a9-b908-4bed-83bc-50bcdbbb9af8.png)
+
+To start using Gatsby you'll need to have a development machine with Node and npm installed then run the following command:
+
+Gatsby has many advantages such as:
+
+* cutting-edge technologies
```
Comment (GuessWhoSamFoo) [inline docs/websites/choose-static-site-generator.md]: Not really.

**Comment by Guaris** [comment]
This guide will not become a part of our library at this time.

**Pull request closed without merging**